src/core/lib/security/credentials/external/external_account_credentials.cc

```diff
@@ -229,10 +230,34 @@ void ExternalAccountCredentials::OnRetrieveSubjectTokenInternal(
   }
 }
 
+bool ValidateUrl(URI url) {
+  absl::string_view scheme(url.scheme());
+  auto string_matcher_scheme =
+      StringMatcher::Create(StringMatcher::Type::kExact, "https");
+  if (!string_matcher_scheme->Match(scheme)) {
+    return false;
+  }
+  std::vector<std::string> v = absl::StrSplit(url.authority(), ':');
```

Comment:
why not just url.host()?

Reply:
There is no api for host() unfortunately.